core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -768,11 +768,14 @@ private[deploy] class Worker(
     }
   }
 
-  private[deploy] def decommissionSelf(): Boolean = {
+  private[deploy] def decommissionSelf(triggeredByWorker: Boolean): Boolean = {
```

Comment:
similarly, there is only one caller side using `triggeredByWorker=true`, we can keep this method unchanged and put the extra code to that caller side.